matkonnerth/nodesetLoader
# [WIP]: Integration test

Working on a simple integration test:
- Implemented a client which connects to a server and prints it's full address space (at the moment only from ObjectsFolder node and only the nodeId - but this will be extended)
- Built a reference server with compiled nodesets (DI and PLCopen at the moment)
- Device under test is a server with NodesetLoader plugin, which imports the same nodesets as the reference server

Test case:
- start reference server, connect with client and print it's addressspace to a file
- start test server, connect with client and print it's address space to a file
- compare both files


Questions:
- Is the test directory location ok? I have put the sources at /backends/open62541/tests/integration. I am not sure if this is a good location. Because I think it's not a unit test, but an integration test
- Test automation? Do you have a suggestion how I should start/stop server/client etc? from CMakeLists file? Execute a shell script? Or can I execute everything with CMake? Or shall I use the unit testing framework?

Thanks for your help

PS: src and CMakeLists files are work in progress
